### PR TITLE
Issue #9631 - update jaspi jar and permissions for Jetty-10

### LINKED ISSUE #9631: Update jaspi jar for jetty-10 and verify permission names
**jetty-10**

Jetty-10 is using a very old jaspi api jar:  `org.eclipse.jetty.orbit:javax.security.auth.message:jar:1.0.0.v201108011116`. Should source a 1.1.3 version of the api (in the `javax` namespace).  

Also, we should double-check the names of some of the permissions that are being checked, and ensure that they align with the specification:  https://download.oracle.com/otn-pub/jcp/jaspic-1_1-mrel2-spec/jaspic-MR1.1Final.pdf?AuthParam=1680790241_7a1577817094a9227c84baf133edccb9

### PR DESCRIPTION
**Pull request by lachlan-roberts**
closes #9631

### CODE REVIEW

File: jetty-jaspi/pom.xml [hunk 1]
```diff
@@ -51,8 +51,9 @@
       <scope>test</scope>
     </dependency>
     <dependency>
-      <groupId>org.eclipse.jetty.orbit</groupId>
-      <artifactId>javax.security.auth.message</artifactId>
+      <groupId>jakarta.security.auth.message</groupId>
```
Comment (olamy) [inline jetty-jaspi/pom.xml]: this change must be updated in top pom and jetty-home pom as well with same coordinate changes

**Review by olamy — CHANGES_REQUESTED** [review]

File: jetty-jaspi/pom.xml (hunk 1 above)
Comment (lachlan-roberts) [reply jetty-jaspi/pom.xml]: thanks, I've fixed the remaining poms now

File: jetty-jaspi/src/main/java/module-info.java [hunk 2]
```diff
@@ -16,7 +16,7 @@
     requires jetty.servlet.api;
     requires org.slf4j;
 
-    requires transitive javax.security.auth.message;
+    requires transitive java.security.auth.message;
```
Comment (janbartel) [inline jetty-jaspi/src/main/java/module-info.java:19]: Are you sure that's right? Jaspi is `javax.security.auth.messge`.

**Review by janbartel — CHANGES_REQUESTED** [review]
Double check the change to module-info.

File: jetty-jaspi/src/main/java/module-info.java (hunk 2 above)
Comment (lachlan-roberts) [reply jetty-jaspi/src/main/java/module-info.java:19]: There is no `module-info.java` in `jakarta.security.auth.message-api-1.1.3.jar` and in the `META-INF/MANIFEST.MF` it says:
```
Automatic-Module-Name: java.security.auth.message
```


You cannot build the project if you use `javax.security.auth.messge` or `jakarta.security.auth.messge`. It says:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.11.0:compile (default-compile) on project jetty-jaspi: Compilation failure
[ERROR] /home/lachlan/webtide/jetty-10.0.x/jetty-jaspi/src/main/java/module-info.java:[19,44] module not found: javax.security.auth.message
```

**Review by olamy — APPROVED** [review]

File: jetty-jaspi/src/main/java/module-info.java (hunk 2 above)
Comment (janbartel) [reply jetty-jaspi/src/main/java/module-info.java:19]: Groan ... that's  a mess.

**Review by janbartel — APPROVED** [review]

**Pull request merged**